### PR TITLE
[DO NOT MERGE UNTIL JOIN STEP HAPPENS][SOAR-282] Delete Exchange Emails With Slack

### PR DESCRIPTION
**Pull request by jmcadams-r7**
## Description: 

This PR adds the Delete Exchange Emails With Slack Workflow

It also renames the directory of Delete O365 Emails With Slack to follow convention

### Assessment
```
rmt-mbp-5357:Delete_Exchange_Emails_With_Slack jmcadams$ icon-validate .
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!
```

#### Workflow Builder
workflow1
workflow2
workflow3

#### Job
image
image
image
image
image
image
image
image

#### Artifact
artifacts1

#### Slack
slack_flow_1

### CODE REVIEW

File: workflows/Delete_Exchange_Emails_With_Slack/help.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Description
+
+This workflow will delete emails matching given search criteria by sending a notification to InsightConnect with Slack. It will look for any emails that match the 'body', 'subject', and 'from'. It can either automatically delete these emails or prompt the user for further action.
+
+# Key Features
+
+* Removes malicious emails using a slack message
```
Comment (mrinehart-r7) [inline workflows/Delete_Exchange_Emails_With_Slack/help.md:7]:
```suggestion
* Removes malicious emails using a Slack message
```

File: workflows/Delete_Exchange_Emails_With_Slack/help.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+# Requirements
+
+The following connections will need to be setup: 
+
+* Microsoft Exchange
```
Comment (mrinehart-r7) [inline workflows/Delete_Exchange_Emails_With_Slack/help.md:13]:
```suggestion
* [Microsoft Exchange](https://insightconnect.help.rapid7.com/docs/microsoft-exchange)
```

File: workflows/Delete_Exchange_Emails_With_Slack/workflow.spec.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+extension: workflow
+products: ["insightconnect"]
+name: Delete_Exchange_Emails_With_Slack
+title: "Delete Exchange Emails With Slack"
+description: "This workflow will delete emails matching given search criteria by sending a notification to InisghtConnect with Slack."
```
Comment (mrinehart-r7) [inline workflows/Delete_Exchange_Emails_With_Slack/workflow.spec.yaml:5]:
```suggestion
description: "This workflow will delete emails matching given search criteria by sending a notification to InsightConnect with Slack."
```

File: workflows/Delete_Exchange_Emails_With_Slack/help.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+`delete-email user@example.com subject="A phishy email" delete=true`
+
+## Technical Details
+
+Plugins utilized by workflow:
+
+|Plugin|Version|Count|
+|----|----|--------|
+|Microsoft Exchange|5.2.0|2|
+|Python 3 Script|2.0.1|2|
```
Comment (mrinehart-r7) [inline workflows/Delete_Exchange_Emails_With_Slack/help.md:49]: Unfortunately we can't use the Python plugins in production workflows. What are you using it for? We may be able to get around it.

Comment (jmcadams-r7) [reply workflows/Delete_Exchange_Emails_With_Slack/help.md:49]: I can't here. It's parsing options from the incoming command.

Comment (jmcadams-r7) [reply workflows/Delete_Exchange_Emails_With_Slack/help.md:49]: Added a new issue to deal with this. We will have to go with python for now. 

See: SOAR-576
https://issues.corp.rapid7.com/browse/SOAR-576

**Pull request merged**